### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by Smart1-hub**
Add redux-thunk to the project.

Use fetch or axios for making API calls.
Fetch book list from Bookstore API on Books page load.
Refactor add book and remove book features to persist changes in the server.
No styling is required.

### CODE REVIEW

File: src/components/BookItem.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { useDispatch } from 'react-redux';
-import { removeBook } from '../redux/Books/books';
+import { deleteBook } from '../redux/Books/books';
 
 const BookItem = ({ book }) => {
   const {
     booktitle, id, author, category,
```
Comment (ShahierNashaat) [inline src/components/BookItem.js]: - [ ] Kindly change this from booktitle to title to make it work and display the title of the book.

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @Smart1-hub , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Smart1-hub — COMMENTED** [review]
'Booktitle' in the BookItem.js has been changed to 'title'

File: src/components/AddNewBook.js [hunk 2]
```diff
@@ -1,27 +1,45 @@
...
+      title,
+      author,
       category,
     };
-    dispatch(addBook(newBook));
+
+    if (newBook.title.trim().length === 0) {
+      setError('Please add Book title to submit...');
+      setInputValues(newBook);
+    } else if (newBook.category === '') {
```
Comment (BenMukebo) [inline src/components/AddNewBook.js:30]: - [ OPTIONAL] Please add also the validation error message to the author field as you did with the title 😉 👍🏻 
- eg: Please add Book author to submit...

**Review by BenMukebo — APPROVED** [review]
### Hi @Smart1-hub, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL] I would  recommend adding the issue link in the contributing sections: 

```
## 🤝 Contributing

Contributions, issues, and feature requests are welcome!
Feel free to check the [issues page](../../issues/).
```
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**